java/src/jmri/managers/AbstractMemoryManager.java

```diff
@@ -110,8 +111,9 @@ public Memory newMemory(@Nonnull String systemName, @CheckForNull String userNam
     }
 
     /** {@inheritDoc} */
+    @Nonnull
     @Override
-    public @Nonnull Memory newMemory(@Nonnull String userName) {
+    public Memory newMemory(@CheckForNull String userName) throws IllegalArgumentException {
         return newMemory(getAutoSystemName(), userName);
     }
```

Comment:
`null` should not be a valid value for the `userName` for this method. The method should first check if userName is null and in that case throw an exception.

Reply:
How else do you express "give me another memory, I don't care what the system name is?"  I guess you could do ` newMemory(getAutoSystemName(), null)` manually....

Reply:
Yes, you are right. I hadn't thought about that case. I will approve the PR.